app/controllers/works_controller.rb

```diff
@@ -0,0 +1,76 @@
+class WorksController < ApplicationController
+  before_action :find_work, only: [:show, :update, :edit, :destroy]
+
+  def index
+    @works = Work.all
+    @albums = Work.sorted_media("album")
+    @books = Work.sorted_media("book")
+    @movies = Work.sorted_media("movie")
+  end
+
+  def show
+    if @work.nil?
+      flash[:error] = "Work has either been deleted or not found."
+      redirect_to root_path
+      return
+    end
```

Comment:
You should move this sort of logic into `find_work` since that way you get this nice behavior everywhere for "free".

Remember, in a controller filter if it produces a response by doing something like `redirect_to` the action will immediately return to the user without running the action's method.